eqs/src/eth_polling.rs

```diff
@@ -277,8 +281,58 @@ impl EthPolling {
                 }
                 CAPEEvents::FaucetInitializedFilter(filter_data) => {
                     let ro_bytes = filter_data.ro_bytes;
-                    let _ro_sol: RecordOpeningSol = AbiDecode::decode(ro_bytes).unwrap();
-                    todo!();
+
+                    // Obtain record opening
+                    let ro_sol: RecordOpeningSol = AbiDecode::decode(ro_bytes).unwrap();
+                    let ro = RecordOpening::from(ro_sol);
+
+                    // Compute record commmitment
+                    let rc = RecordCommitment::from(&ro);
+
+                    // Compute memo
+                    let mut rng = ChaChaRng::from_entropy();
+                    let memo = ReceiverMemo::from_ro(&mut rng, &ro, &[]).unwrap();
+
+                    // Update the state
+                    let faucet_transition = Faucet {
+                        memo: Box::new(memo.clone()),
+                        ro: Box::new(ro),
+                    };
+
+                    self.pending_commit_event.push(faucet_transition);
```

Comment:
I think we we have a small inconsistency here, in the sense that we use pending_commit_event as an accumulator between events that get made available to the wallet (e.g. erc20_deposited), but this is actually something that will be used to generate a `LedgerEvent::Memos` **now**. I think, from the wallet's perspective, this event is not visible.

Remove this line and the block above, we'll add them back if we add a raw_events endpoint. 

Otherwise, this looks good.

Reply:
Right, I was unsure about adding this *transition*. The code has been removed in 6dce87f.